src/type-checker.cc

```diff
@@ -741,7 +766,7 @@ Result TypeChecker::OnRefNullExpr(Type type) {
 Result TypeChecker::OnRefIsNullExpr() {
   Type type;
   Result result = PeekType(0, &type);
-  if (!type.IsRef()) {
+  if (!type.IsFuncOrExternRef()) {
```

Comment:
Why can't OnRefIsNullExpr work for refs?   I that is to say why not use `IsRef` here?

Reply:
fixed